File: src/main/java/net/sharplab/epubtranslator/core/model/FileEntry.java
package net.sharplab.epubtranslator.core.model;


import net.sharplab.epubtranslator.core.util.ArrayUtil;

public class FileEntry {

    private String name;
    private byte[] data;

    public FileEntry(String name, byte[] data){
        this.name = name;
        this.data = ArrayUtil.clone(data);
    }

    public String getName() {
        return name;
    }

    public byte[] getData() {
        return ArrayUtil.clone(data);
    }
}

File: src/main/java/net/sharplab/epubtranslator/core/model/FileEntry.kt
package net.sharplab.epubtranslator.core.model

import net.sharplab.epubtranslator.core.util.ArrayUtil

class FileEntry(val name: String, data: ByteArray) {

    val data: ByteArray = ArrayUtil.clone(data)
        get() = ArrayUtil.clone(field)

}